internal/server/data/identity.go

```diff
@@ -107,38 +109,37 @@ func AssignIdentityToGroups(tx WriteTxn, user *models.Identity, provider *models
 		if !found {
 			group := &models.Group{
 				Name:              name,
-				CreatedByProvider: provider.ID,
+				CreatedByProvider: user.ProviderID,
 			}
 
 			if err = CreateGroup(tx, group); err != nil {
-				return fmt.Errorf("create group: %w", err)
+				return nil, fmt.Errorf("create group: %w", err)
 			}
 			groupID = group.ID
 		}
 
-		rows, err := tx.Query("SELECT identity_id FROM identities_groups WHERE identity_id = ? AND group_id = ?", user.ID, groupID)
+		rows, err := tx.Query("SELECT identity_id FROM identities_groups WHERE identity_id = ? AND group_id = ?", user.IdentityID, groupID)
 		if err != nil {
-			return err
+			return nil, err
 		}
 		ids, err := scanRows(rows, func(item *uid.ID) []any {
 			return []any{item}
 		})
 		if err != nil {
-			return err
+			return nil, err
 		}
 
 		if len(ids) == 0 {
 			// add user to group
-			_, err = tx.Exec("INSERT INTO identities_groups (identity_id, group_id) VALUES (?, ?)", user.ID, groupID)
+			_, err = tx.Exec("INSERT INTO identities_groups (identity_id, group_id) VALUES (?, ?)", user.IdentityID, groupID)
 			if err != nil {
-				return fmt.Errorf("insert: %w", handleError(err))
+				return nil, fmt.Errorf("insert: %w", handleError(err))
 			}
+			identity.Groups = append(identity.Groups, models.Group{Model: models.Model{ID: groupID}, OrganizationMember: models.OrganizationMember{OrganizationID: identity.OrganizationID}, Name: name})
```

Comment:
This line would read a lot better if it were formatted with each field of the struct on a separate line.